polkadot/xcm/src/lib.rs

```diff
@@ -248,7 +404,7 @@ versioned_type! {
 	pub enum VersionedNetworkId {
 		#[codec(index = 2)]
 		V2(v2::NetworkId),
-		#[codec(index = 3)]
+		#[codec(index = 3)] // Same for v4
```

Comment:
I think if it's the same for V4, we'd still want to have a V4 variant here to make sure the versioned type matches the XCM version that we're using.

Reply:
I had some issues with multiple implementations on the same type since the underlying type is the same, just reexported.
It does sound reasonable though, so I'll try to fix it.